### PR TITLE
fix: blacklist "none" as a possible environment name

### PR DESCRIPTION
**Pull request by wmak**
- This is because there's an app [function](https://github.com/getsentry/sentry/blob/master/src/sentry/models/environment.py#L107) that uses the string "none" from a URL path segment (case-sensitive) to represent the empty string environment name.

### CODE REVIEW

File: general/src/store/normalize.rs [hunk 1]
```diff
@@ -65,6 +65,10 @@ pub fn is_valid_platform(platform: &str) -> bool {
     VALID_PLATFORMS.contains(&platform)
 }
 
+pub fn is_valid_environment(environment: &str) -> bool {
```
Comment (dashed) [inline general/src/store/normalize.rs:68]: Wondering if this function should be in parity with this one: https://github.com/getsentry/sentry/blob/3fd6386943efb8a5737e0e471ba85998321c18f1/src/sentry/models/environment.py#L45-L54

this was introduced in https://github.com/getsentry/sentry/pull/9815

Comment (jan-auer) [reply general/src/store/normalize.rs:68]: @dashed good point. This part is handled here:
https://github.com/getsentry/semaphore/blob/b95b6a2dec2bf611d610b66f6df2543756b5be49/general/src/protocol/event.rs#L285-L287

**Comment by jan-auer** [comment]
LGTM, thanks! 

@wmak if there's a place in getsentry/sentry where we do similar checks, please consider to reference the `INVALID_ENVIRONMENTS` constant here, so that future travellers can keep it in sync :)

**Review by jan-auer — APPROVED** [review]

File: general/src/store/normalize.rs (hunk 1 above)
Comment (dashed) [reply general/src/store/normalize.rs:68]: Gotcha thanks 👍

**Pull request merged**